certif/app/routes/application.js

```diff
@@ -1,53 +1,13 @@
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
 
-const FRENCH_LOCALE = 'fr-FR';
-
 export default class ApplicationRoute extends Route {
-  @service currentUser;
   @service featureToggles;
-  @service intl;
-  @service currentDomain;
-  @service dayjs;
-  @service locale;
+  @service session;
 
   async beforeModel(transition) {
     await this.featureToggles.load();
-    const locale = transition.to.queryParams.lang;
-    await this.handleLocale(locale);
-    return this._loadCurrentUser();
-  }
-
-  _loadCurrentUser() {
-    return this.currentUser.load();
-  }
-
-  async handleLocale(localeFromQueryParam) {
-    const domain = this.currentDomain.getExtension();
-    const defaultLocale = 'fr';
-    const domainFr = 'fr';
-
-    if (domain === domainFr) {
-      this._setLocale(domainFr);
-
-      if (!this.locale.hasLocaleCookie()) {
-        this.locale.setLocaleCookie(FRENCH_LOCALE);
-      }
-
-      return;
-    }
-
-    if (localeFromQueryParam) {
-      this._setLocale(localeFromQueryParam);
-    } else {
-      this._setLocale(defaultLocale);
-    }
-  }
-
-  _setLocale(locale) {
-    const defaultLocale = 'fr';
-    this.intl.setLocale([locale, defaultLocale]);
-    this.dayjs.setLocale(locale);
-    this.dayjs.self.locale(locale);
+    const localeFromQueryParam = transition.to.queryParams.lang;
+    await this.session.loadCurrentUserAndSetLocale(localeFromQueryParam);
```

Comment:
```suggestion
    await this.session.loadCurrentUser(); // avant la récupération de la langue dans l'URL
    await this.session.handleLocaleFromQueryParam(localeFromQueryParam);
```

Je serais plus pour séparer la logique. Je pense que cela facilitera les tests.

Reply:
Je comprends.

Par contre cette méthode nous ne voudrions pas de ce nommage : 
`await this.session.handleLocaleFromQueryParam(localeFromQueryParam);`
mais plutôt de ce nommage : 
`await this.session.handleLocale(localeFromQueryParam);`

En effet ce sont de multiples informations qui sont utilisées (localeFromQueryParam, TLD, utilisateur connecté ou non, locale de l'utilisateur et locale par défaut) pour déterminer au final la locale qui sera positionnée pour l'appli Ember.

Reply:
+1 pour faire un refacto, y'a ce ticket PIX-7570 dans le backlog dette technique, je propose de le faire en dehors de cette PR

Reply:
La récupération du user est maintenant gérée à l'extérieur de la méthode `handleLocale` (anciennement nommée `handleLocaleFromQueryParam`). Merci.